### DISCUSSION TITLE
Anti script kiddie measures

### DISCUSSION THREAD

**Discussion by C3rb3rus-666** [termux/termux-app#2261] [Q&A] [answered]
Have there been any plans for anti-kiddie measures?
It is a bit sad that a great development like termux is used in bad things by hundreds of groups in wsp and Facebook .
Although it is always to highlight the great work of this great community behind termux make us life easier for many developers I in my case I use my utilities for my work or sharing files with my colleagues in the same field thanks for this great project.

**Comment by agnostic-apollo** [comment, marked as answer]
Removing hacking/phishing related packages and banning any such discussion/questions on official termux support forums is the only thing that can be done by us. Outside official termux repos/forums, users are free to do whatever they want, as they should.

https://wiki.termux.com/wiki/Hacking

**Comment by ghost** [comment]
Enforcing community rules and moderation as well as removing "non-grata" packages is enough and is something that is under our control.

Doing anything more (e.g. locking certain commands) is possible, but will seriously interfere with legit use. Our community is wide and besides skids there are real penetration testers. People should be able to install and use package even if it is not available in our repository.

**Comment by Aridlvory** [comment]
Well it's not possible.... Removing repos are the only thing the termux people can do
Hacking can be done like running a python program on termux.... So tracking and stopping those are somewhat impossible
Cyber police might find em.....
But we can't stop em running python scripts, and we cannot identify if the py script is good or bad
(Not only py tho)

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: C3rb3rus-666 ***@***.***>
Sent: Wednesday, September 8, 2021 1:43:46 PM
To: termux/termux-app ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [termux/termux-app] Anti script kiddie measures (#2261)


Have there been any plans for anti-kiddie measures?
It is a bit sad that a great development like termux is used in bad things by hundreds of groups in wsp and Facebook .
Although it is always to highlight the great work of this great community behind termux make us life easier for many developers I in my case I use my utilities for my work or sharing files with my colleagues in the same field thanks for this great project.

**Reply by ghost** [reply]
> But we can't stop em running python scripts

That's not correct. Many of tools are common and have very specific names and in case of scripts, they often have specific patterns. Patching interpreter to check the file name and do certain actions when file name is `msfconsole`, `wifite`, `snoop.py` and so on is possible.

The problem is, as I wrote in https://github.com/termux/termux-app/discussions/2261#discussioncomment-1295097, this would be overdrive and will interfere with legit use of application. Termux is finally a general purpose app and is not (and shouldn't be) locked down only to certain features.
